argocd/resource_argocd_application.go

```diff
@@ -92,11 +93,15 @@ func resourceArgoCDApplicationCreate(d *schema.ResourceData, meta interface{}) e
 	}
 	d.SetId(app.Name)
 	if wait, ok := d.GetOk("wait"); ok && wait.(bool) {
-		return resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+		err = resource.RetryContext(context.Background(), d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
 			a, err := c.Get(context.Background(), &applicationClient.ApplicationQuery{
 				Name: &app.Name,
 			})
 			if err != nil {
+				fmt.Printf("%v\n", err)
+				if strings.Contains(err.Error(), "NotFound") {
```

Comment:
Can you detail that change of behaviour?

Reply:
I think we should retry waiting as long as the resource is not found. If another type of error occurs, the wait-loop should stop there.

Reply:
That use case should not happen as it is already handled in line 91 to 93, therefore I don't think it's necessary to add that check.